server/src/main/java/org/elasticsearch/index/query/CoordinatorRewriteContextProvider.java

```diff
@@ -74,7 +74,8 @@ public CoordinatorRewriteContext getCoordinatorRewriteContext(Index index) {
             parserConfig,
             client,
             nowInMillis,
-            new DateFieldRangeInfo(timestampFieldType, timestampRange, dateFieldRangeInfo.eventIngestedFieldType(), eventIngestedRange)
+            new DateFieldRangeInfo(timestampFieldType, timestampRange, dateFieldRangeInfo.eventIngestedFieldType(), eventIngestedRange),
+            indexMetadata.getTierPreference().isEmpty() == false ? indexMetadata.getTierPreference().getFirst() : ""
```

Comment:
We are potentially artificially limiting the coordinator tier rewrite to [only indices that have the time range configured](https://github.com/elastic/elasticsearch/pull/114990/files#diff-0ad1800461d83a801b07a1207f4395615a0e9b6b5aad9ded33bda88e0f784d47R56) (partially mounted indices). We don't have to have this limitation for the `_tier` coordination rewrite however, the current implementation is sufficient for skipping frozen (where all indices do indeed have the `@timestamp` and `event.ingested` ranges). 

In the interest of keeping this fix targeted and contained I believe it's a good compromise for now but curious what you think.

Reply:
That's quite subtle, is that because above we return null if the range field info isn't found in the cluster state? 

```
DateFieldRangeInfo dateFieldRangeInfo = mappingSupplier.apply(index);
if (dateFieldRangeInfo == null) {
    return null;
}
```

I think this is ok, but perhaps let's clarify in the code that it's by design and not by mistake? Perhaps we will want to extend this in the future to all indices? I mean there is no technical reason that this tier filtering cannot be extended to all indices, provided that the tier field cannot ever change (which may happen for some constant field types - e.g. they may be unset and get set for the first time via dynamic mappings)

Reply:
now that this is merged, do we want to do the follow-up to allow filtering on tier to work on all indices?

Reply:
Yes, PR is up https://github.com/elastic/elasticsearch/pull/115797